### ISSUE TITLE
Api.ai - Empty query parameter

### ISSUE THREAD

**Issue by ribb3r** [mpociot/botman#204]
Hi,

I'm just trying to implement the Api.ai middleware. For some reason, the query parameter is empty during POST request to Api.ai (I think it should be the user message):

_User says: This is my test_

```
$botman->hears('myAction', function(BotMan $bot) {
    $bot->reply("this is my reply");
})->middleware(ApiAi::create('xxx')->listenForAction());
```

`{"query":[""],"sessionId":1487019928,"lang":"en"}`

Did I miss something?

Thank you!

**Comment by thedotedge** [comment]
Which Driver do you use? Query should be just `$message->getMessage()`, are you getting the message at https://github.com/mpociot/botman/blob/master/src/Mpociot/BotMan/Middleware/ApiAi.php#L67?

**Comment by ribb3r** [comment]
I'm using the FacebookDriver. Yes, this is the line where I get the message.

**Comment by thedotedge** [comment]
If you remove the middleware and echo the received message back to the user, does it work?
Any other settings you apply to `Botman`?

**Comment by ribb3r** [comment]
I've added two custom drivers and I'm using DoctrineCache. I've also removed both options to simulate botmans "default" behaviour. Without success.

Could you give me an example for:
> remove the middleware and echo the received message back to the user

Thank you.

**Comment by thedotedge** [comment]
Something like:
```
$botman->hears('hello', function(BotMan $bot) {
    $bot->reply($bot->getMessage());
})
```

**Comment by ribb3r** [comment]
You're right. It seems to be a general issue and not related to the Api.ai middleware. I did:

```
$botman->hears('hello', function(BotMan $bot) {
       $bot->reply(' . ' . json_encode($bot->getMessage()));
});
```

And got:
`. {}`

So `$bot->getMessage()` is empty. Any advice?

Thank you.

**Comment by mpociot** [comment]
@ribb3r `$bot->getMessage()` is not empty. It's just not JSON serializable.

Try this instead and you should receive a dump of the message object:

```php
$botman->hears('hello', function(BotMan $bot) {
    $bot->reply(' . ' . print_r($bot->getMessage(), true));
});
```

**Comment by ribb3r** [comment]
makes sense ;-) Thank you.

My issue was resolved with  108c890

**Comment by mpociot** [comment]
Awesome :)
